jcl/src/java.base/share/classes/java/security/AccessController.java

```diff
@@ -485,7 +486,12 @@ private static AccessControlContext getContextHelper(boolean forDoPrivilegedWith
 			accTmp = new AccessControlContext(pDomains, newAuthorizedState);
 		}
 		if (null != acc && null != acc.domainCombiner) {
-			accTmp.domainCombiner = acc.domainCombiner;
+			if (dc == null) {
+				// this dc will be set to accContext.domainCombiner
+				dc = acc.domainCombiner;
+			} else {
+				accTmp.domainCombiner = acc.domainCombiner;
```

Comment:
What I'm wondering is if the accTmp.domainCombiner, other than the one used to assign accContext, is ever used for anything? Just in terms of riskiness for this change, I liked the original code (without the `if (dc == null) ... else ...` better since this version could leave a chained nextStackAcc domainCombiner as null, when before this PR it had a value.

Reply:
When there are multiple `AccessControlContext` with `domainCombiner` objects discovered during stack walking, only one can be set to returning `AccessControlContext`. It doesn't appear those `domainCombiner` within `AccessControlContext.nextStackAcc` are used. 
For now I would leave it as is to prevent any behaviour changes.
Reverted to original version. Please have another look.